crates/ruff/src/args.rs

```diff
@@ -28,6 +28,15 @@ use ruff_workspace::configuration::{Configuration, RuleSelection};
 use ruff_workspace::options::{Options, PycodestyleOptions};
 use ruff_workspace::resolver::ConfigurationTransformer;
 
+/// All configuration options that can be passed "globally",
+/// i.e., can be passed to all subcommands
+#[derive(Debug, Default, Clone)]
+pub struct GlobalConfigArgs {
+    pub log_level: LogLevel,
```

Comment:
What's the relation between `log_level` and `Args::log_level_args`. How do we ensure that the two are in sync?

Reply:
`log_level_args` is an instance of the `LogLevelArgs` struct that contains information about the various logging-related flags the user may or may not have passed on the CLI.

`log_level` is a variant of the `LogLevel` enum: the flags have now been parsed internally so that we now know what logging level to apply henceforth.

The `LogLevelArgs` struct is parsed into a `LogLevel` variant in `Args::partition()`

Reply:
it's quite nice to eagerly parse `LogLevelArgs` into more structured data, as I'm doing now, but I think that might be trickier if we use `flatten` like you suggest in https://github.com/astral-sh/ruff/pull/10150#discussion_r1507492780

Reply:
Oh I see, `Args`  has a `partition` too and `GlobalConfigArgs` (as of today) is not really a `Args` but more an `Arguments` (it's not populated by clap).

I think we can make `log_level` a function on `GlobalConfigArgs` and inline `LogLevelArgs` into `GlobalConfigArgs`